packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -84,6 +155,92 @@ export class Table extends Construct {
         return errors;
     }
 
+    private validateAutoScalingProps(props: AutoScalingProps) {
+        if (props.targetValue < 10 || props.targetValue > 90) {
+            throw new RangeError("scalingTargetValue for predefined metric type DynamoDBReadCapacityUtilization/"
+                + "DynamoDBWriteCapacityUtilization must be between 10 and 90; Provided value is: " + props.targetValue);
+        }
+        if (props.scaleInCooldown < 0) {
+            throw new RangeError("scaleInCooldown must be greater than or equal to 0; Provided value is: " + props.scaleInCooldown);
+        }
+        if (props.scaleOutCooldown < 0) {
+            throw new RangeError("scaleOutCooldown must be greater than or equal to 0; Provided value is: " + props.scaleOutCooldown);
+        }
+        if (props.maxCapacity < 0) {
+            throw new RangeError("maximumCapacity must be greater than or equal to 0; Provided value is: " + props.maxCapacity);
+        }
+        if (props.minCapacity < 0) {
+            throw new RangeError("minimumCapacity must be greater than or equal to 0; Provided value is: " + props.minCapacity);
+        }
+    }
+
+    private buildAutoScaling(scalingPolicyResource: applicationautoscaling.ScalingPolicyResource | undefined,
+                             scalingType: string,
+                             props: AutoScalingProps) {
+        if (scalingPolicyResource) {
+            throw new Error(`${scalingType} Auto Scaling already defined for Table`);
+        }
+
+        this.validateAutoScalingProps(props);
+        const autoScalingRole = this.buildAutoScalingRole(`${scalingType}AutoScalingRole`);
+
+        const scalableTargetResource = new applicationautoscaling.ScalableTargetResource(
+            this, `${scalingType}CapacityScalableTarget`, this.buildScalableTargetResourceProps(
+                `dynamodb:table:${scalingType}CapacityUnits`, autoScalingRole, props));
+
+        return new applicationautoscaling.ScalingPolicyResource(
+            this, `${scalingType}CapacityScalingPolicy`,
+            this.buildScalingPolicyResourceProps(`DynamoDB${scalingType}CapacityUtilization`, `${scalingType}Capacity`,
+            scalableTargetResource, props));
+    }
+
+    private buildAutoScalingRole(roleResourceName: string) {
```

Comment:
Move all the helper methods is inner functions of `buildAutoScaling`

Reply:
is this considered superior design in JS/TS? because I see this as writing a big method in other programming languages, and I'm a firm believer in small methods. i understand that you are asking to make an inner function and not copy the code but and at the same time we want to ensure these methods are not called by anyone outside of buildAutoScaling. but since all these methods are private is that a high risk?